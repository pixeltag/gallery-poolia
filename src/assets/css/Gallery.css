.gallery-container {
  max-width: 1280px;
  margin: 0 auto;
  column-count: 4;
  -webkit-column-count: 4;
  -moz-column-count: 4;
  column-gap: 10px;
  -webkit-column-gap: 10px;
  -moz-column-gap: 10px;
}

.img-figure {
  position: relative;
  border: 1px solid #ededed;
  display: inline-block;
  margin-bottom: 10px;
  width: 100%;
  text-align: center;
  cursor: pointer;
  transition: all 0.3s ease;
}

.img-figure:hover {
  box-shadow: 0px 3px 6px 0px #ccc;
  -webkit-box-shadow: 0px 3px 6px 0px #ccc;
  -moz-box-shadow: 0px 3px 6px 0px #ccc;
}

@media (max-width: 992px) {
  .gallery-container {
    column-count: 3;
    -webkit-column-count: 3;
    -moz-column-count: 3;
  }
}

@media (max-width: 768px) {
  .gallery-container {
    column-count: 2;
    -webkit-column-count: 2;
    -moz-column-count: 2;
  }
}



.title-figure {
  margin: 5px;
  line-height: 1.5;
  margin-bottom: 5px;
  font-size: 15px;
}

/* img modal */

.modal {
  width: 100%;
  height: 100vh;
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #000;
  transition: opacity 0.6s ease, visibility 0.2s ease,
    transform 0.3s ease-in-out;
  visibility: hidden;
  opacity: 0;
  transform: scale(0);
  overflow: hidden;
  z-index: 99;
  flex-direction: column;
}
.modal.show {
  visibility: visible;
  opacity: 1;
  transform: scale(1);
}

.modal-body {
    color: #fff;
    max-width: 1280px;
    position: fixed;
    bottom: 0;
    padding: 1.5em;
    background: rgba(0,0,0,0.5);
}
.modal-body h3 {
  border-bottom: 1px solid #282828;
  margin-bottom: 10px;
  font-size: 20px;
}

.modal-body p {
  font-size: 14px;
  color: #ccc;
}

.close-btn {
  position: fixed;
  top: 20px;
  right: 20px;
  cursor: pointer;
  background: transparent;
  border: 0;
}


@media (max-width: 576px) {
  .gallery-container {
    column-count: 1;
    -webkit-column-count: 1;
    -moz-column-count: 1;
  }

  .modal-img {
    width: 100%;
  }

  .modal {
    justify-content: start;
  }
}