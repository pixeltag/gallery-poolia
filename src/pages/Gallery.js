import React, { useState, useEffect, useCallback } from "react";
import "../assets/css/Gallery.css";
import { fetchImages } from "../libs/utils";
import ImgFigure from "../components/ImgFigure";
import ImgModal from "../components/ImgModal";

const Gallery = () => {
  const [imgData, setImgData] = useState([]);
  const [imgPreview, setImgPreview] = useState(null);
  const [modal, setModal] = useState(false);

  useEffect(() => {
    // get data from img service
    fetchImagesData();
  }, []);

  const fetchImagesData = useCallback(async () => {
    const response = await fetchImages();
    setImgData(response.images);
  }, []);

  useEffect(() =>  document.addEventListener("keydown", escapePressHandler, false));

  const openModal = (data) => {
    setImgPreview(data);
    setModal(true);
  };

  const closeModal = () => {
    setImgPreview(null);
    setModal(false);
  };

  const escapePressHandler = useCallback((e) => {
    if (e.keyCode === 27) {
      closeModal();
    }
  }, []);

  return (
    <React.Fragment>
      <section className="gallery-container">
        <h2 className="hide">Gallery Section</h2>
        {imgData.length > 0 ? (
          <div data-testid="list">
            {
              imgData.map((img, index) => (
              <ImgFigure {...img} key={index} openModalCallback={openModal} />
              ))
            }
          </div>
        ) : (
          <div className="not-found">Loading</div>
        )}
      </section>

      <ImgModal
        {...imgPreview}
        modalStatus={modal}
        handleCloseModalCb={closeModal}
      />
    </React.Fragment>
  );
};

export default Gallery;
