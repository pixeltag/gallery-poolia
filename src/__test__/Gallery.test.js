import {
  render,
  screen,
  cleanup,
  waitFor,
} from "@testing-library/react";

import { shallow, configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { unmountComponentAtNode } from "react-dom";
import Gallery from "./../pages/Gallery";

configure({ adapter: new Adapter() });

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  cleanup();
  container.remove();
  container = null;
});

test("Gallery component renders correctly and has section text", async () => {
  const wrapper = shallow(<Gallery />);
  expect(await wrapper.find("Gallery Section")).toBeTruthy();
  cleanup()
});

test("Gallery component should match the snapshot", () => {
  const wrapper = shallow(<Gallery />);
  expect(wrapper.html()).toMatchSnapshot();
  cleanup();
});

it("should have an email field", () => {
  const wrapper = shallow(<Gallery />);
  expect(wrapper.find(".gallery-container").length).toEqual(1);
  cleanup();
});

test("when fetch data loading is showing", async () => {
  render(<Gallery />, container);
  expect(await screen.findByText(/Loading/i)).toBeInTheDocument();
  cleanup();
});

test("fetchImages has been called once when component initial", async () => {
  render(<Gallery />, container);

  expect(await screen.findByText("Loading")).toBeInTheDocument();

  expect(
    await waitFor(() => screen.findByTestId("list", {}, { timeout: 3000 }))
  ).toBeInTheDocument();
});


test("child's callback function fire open modal fn", async () => {
  const openModalMocked = jest.fn()
  const wrapper = shallow(<Gallery />);
  const child =  await waitFor(() => wrapper.find("ImgFigure"))
  child.prop.openModalCallback = openModalMocked;
  child.prop.openModalCallback()
  expect(openModalMocked).toHaveBeenCalled();
  cleanup();
});