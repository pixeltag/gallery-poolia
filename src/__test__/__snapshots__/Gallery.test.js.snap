// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery component should match the snapshot 1`] = `"<section class=\\"gallery-container\\"><h2 class=\\"hide\\">Gallery Section</h2><div class=\\"not-found\\">Loading</div></section><div class=\\"modal\\"><img src=\\"close.svg\\" alt=\\"Close modal\\" class=\\"close-btn\\"/><img class=\\"modal-img\\"/><div class=\\"modal-body\\"><h3>Image title</h3><p></p></div></div>"`;
